addons/purchase_stock/models/purchase.py

```diff
@@ -468,7 +468,7 @@ def _prepare_stock_move_vals(self, picking, price_unit, product_uom_qty, product
             'picking_type_id': self.order_id.picking_type_id.id,
             'group_id': self.order_id.group_id.id,
             'origin': self.order_id.name,
-            'propagate_date': self.propagate_date,
+            'propagate_date': self.order_id.picking_type_id.warehouse_id.buy_pull_id.propagate_date,
```

Comment:
i wonder if we shouldn't only use this logic when there is no move_dest_ids ?
the original issue is if i create a PO manually, propaate_date is always false
but if it comes from an mto chain, it should use the one from the procurement group and with your patch it won't be the case

i'm not sure of the proper fix but i suggest
``` python
self.order_id.picking_type_id.warehouse_id.buy_pull_id.propagate_date if not self.move_dest_ids else self.propagate_date
```